spec/controllers/admin/promotion/codes_controller_spec.rb

```diff
@@ -0,0 +1,54 @@
+require 'spec_helper'
+
+describe Spree::Admin::Promotion::CodesController, type: :controller do
+  stub_authorization!
+
+  describe '#index' do
+    let(:user) { mock_model(Spree::User, id: 1) }
+    let(:promotion) { mock_model(Spree::Promotion, id: 1) }
+    let(:code) { mock_model(Spree::Promotion::Code, user: user, promotion: promotion) }
+    let(:codes) { [code] }
+
+    before do
+      allow(controller).to receive(:spree_current_user).and_return(user)
+      allow(Spree::Promotion).to receive(:find_by).and_return(promotion)
+    end
+
+    def send_request(params = {})
+      spree_get :index, params
+    end
+
+    context 'when promotion found' do
+      it 'expects to assign promotion' do
+        send_request(promotion_id: promotion.id)
```

Comment:
@AnuragBhardwaj1 Add `promotion_id: promotion.id` to send_request itself as being used for all test cases.